### DISCUSSION TITLE
Newb question for API Keys

### DISCUSSION THREAD

**Discussion by lantau22** [OpenBB-finance/OpenBB#331] [Q&A] [answered]
Hello, 

I don’t have any relevant programming experience, but was able to get the GS Terminal up and running.  I now want to enable some of the API Calls/Keys, for example Quandl.  

I setup an account in Quandl and now have an API key.  Using notepad, I edited the “config_terminal” file.  For API_KEY_QUANDL, I went to “REPLACE_ME” and changed it to my new API Key and saved the file.

However, when now I launch python terminal.py, I get a syntax error.

What is the correct way to use my new API Key?   Thanks!

**Comment by aia** [comment, marked as answer]
If you are editing a file, make sure that your key is in quotes.
The syntax error that you are seeing is probably because you have an incorrect python formatting.

Another way to specify a key is to pass it in an environment variable.
When you put your key in an environment variable, you don't have to edit any python files.

**Comment by lantau22** [comment]
In notepad, I replaced the “ReadMe” (just the text, not the quotes) with
the 20 digit API key.  Notepad should be ok as an editor correct?

Can you give me an example of how I would pass the API key in an
environment variable?  Again, I really don’t have any programming
experience other than some stuff 20 years ago (that I honestly don’t
remember).  Thanks!

On Mon, Apr 12, 2021 at 13:21 Artem Veremey ***@***.***>
wrote:

> If you are editing a file, make sure that your key is in quotes.
> The syntax error that you are seeing is probably because you have an
> incorrect python formatting.
>
> Another way to specify a key is to pass it in an environment variable.
> When you put your key in an environment variable, you don't have to edit
> any python files.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/DidierRLopes/GamestonkTerminal/discussions/331#discussioncomment-598051>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATUK22G646CNCLTCGLZP4GLTIJ7PDANCNFSM42YPGRYQ>
> .
>
-- 
Sent from Gmail Mobile

**Reply by aia** [reply]
On Mac or Linux you export environment variables like so:

```
export GT_API_KEY_QUANDL="xxx"

python terminal.py
```

On Windows you export environment variables in the following way:

```
set GT_API_KEY_QUANDL=xxx

python terminal.py
```

**Comment by lantau22** [comment]
Update:

For whatever reason, I attempted the “REPLACE_ME” edit one more time with
the API key and it worked...  No idea what I screwed up the first time!

On Mon, Apr 12, 2021 at 14:08 Sam Trask ***@***.***> wrote:

> In notepad, I replaced the “ReadMe” (just the text, not the quotes) with
> the 20 digit API key.  Notepad should be ok as an editor correct?
>
> Can you give me an example of how I would pass the API key in an
> environment variable?  Again, I really don’t have any programming
> experience other than some stuff 20 years ago (that I honestly don’t
> remember).  Thanks!
>
> On Mon, Apr 12, 2021 at 13:21 Artem Veremey ***@***.***>
> wrote:
>
>> If you are editing a file, make sure that your key is in quotes.
>> The syntax error that you are seeing is probably because you have an
>> incorrect python formatting.
>>
>> Another way to specify a key is to pass it in an environment variable.
>> When you put your key in an environment variable, you don't have to edit
>> any python files.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/DidierRLopes/GamestonkTerminal/discussions/331#discussioncomment-598051>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ATUK22G646CNCLTCGLZP4GLTIJ7PDANCNFSM42YPGRYQ>
>> .
>>
> --
> Sent from Gmail Mobile
>
-- 
Sent from Gmail Mobile

**Comment by lantau22** [comment]
Ok, easy enough.  Thank you!

On Mon, Apr 12, 2021 at 14:43 Artem Veremey ***@***.***>
wrote:

> On Mac or Linux you export environment variables like so:
>
> export GT_API_KEY_QUANDL="xxx"
>
> python terminal.py
>
> On Windows you export environment variables in the following way:
>
> set GT_API_KEY_QUANDL=xxx
>
> python terminal.py
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/DidierRLopes/GamestonkTerminal/discussions/331#discussioncomment-598229>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATUK22AC33WNXYH57ZMUOYTTIKJCRANCNFSM42YPGRYQ>
> .
>
-- 
Sent from Gmail Mobile
